### PR TITLE
Add desktop build scripts

### PR DESCRIPTION
**Pull request by vkolesnikov-intermedia**
Build scripts for java-desktop: linux, mac arm64, mac x86.
Output binaries will allow create pjsip-client instance in own java process and interact with it using RMI. It will be used for integration tests for java/android platform.

### CODE REVIEW

File: .github/workflows/linux-release.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+on:
+  push:
+    tags:
+      - '*'
+
+env:
+  RELEASE_ARCHIVE_NAME: linux-release
+
+jobs:
```
Comment (vkolesnikov-intermedia) [inline .github/workflows/linux-release.yml:11]: Maybe someone knows how reuse jobs body code? I found composite actions, but it seems that own repository for them needed, and it too much for this case.

File: .github/workflows/macos-arm64-release.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  push:
+    tags:
+      - '*'
+
+env:
+  RELEASE_ARCHIVE_NAME: macOS-arm64-release
+
+jobs:
+  build-macos-arm64:
+    runs-on: macos-14
```
Comment (vkolesnikov-intermedia) [inline .github/workflows/macos-arm64-release.yml:13]: I found that macos-latest can provide runner with less version, so to build on arm I use certain. Maybe there is some convenient way to specify arch? I didn't found explicit way in workflow docs. 
It must be handle in future: using cross-compilation (according docs of pjsip it is experimental), or just keep hardcoded version and increase it if needed

File: build-scripts/build [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+function setConfigSite {
+  echo "#include <pj/config_site_sample.h>" >> "$CONFIG_SITE_PATH"
+  echo "#define PJMEDIA_HAS_VIDEO 0" >> "$CONFIG_SITE_PATH"
+  echo "#define PJMEDIA_AUDIO_DEV_HAS_ANDROID_JNI 0" >> "$CONFIG_SITE_PATH"
+  echo "#define PJMEDIA_AUDIO_DEV_HAS_OPENSL 0" >> "$CONFIG_SITE_PATH"
+  echo "#define PJSIP_AUTH_AUTO_SEND_NEXT 0" >> "$CONFIG_SITE_PATH"
+  echo "#define PJMEDIA_HAS_OPUS_CODEC 1" >> "$CONFIG_SITE_PATH"
+  echo "#define PJSUA_DETECT_MERGED_REQUESTS 0" >> "$CONFIG_SITE_PATH"
+
+  if [ "${ENABLE_IPV6}" == "1" ]
```
Comment (anikitin-intermedia) [inline build-scripts/build:62]: This should be enabled always

File: build-scripts/build [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  cp -r "pjlib/include"/* "$includeDir/pjlib"
+
+  mkdir -p "$includeDir/pjlib-util"
+  cp -r "pjlib-util/include"/* "$includeDir/pjlib-util"
+
+  mkdir -p "$includeDir/pjmedia"
+  cp -r "pjmedia/include"/* "$includeDir/pjmedia"
+
+  mkdir -p "$includeDir/pjnath"
+  cp -r "pjnath/include"/* "$includeDir/pjnath"
```
Comment (anikitin-intermedia) [inline build-scripts/build:102]: Please, don't forget about 'pjnat64'

Comment (vkolesnikov-intermedia) [reply build-scripts/build:97]: fxd

File: build-scripts/build (hunk 3 above)
Comment (vkolesnikov-intermedia) [reply build-scripts/build]: fxd

**Comment by vkolesnikov-intermedia** [comment]
if you want I can represent in naming java specific or make scripts more modular to construct build with swig/without/with something else

File: .github/workflows/linux-release.yml (hunk 1 above)
Comment (inekrasov-intermedia) [reply .github/workflows/linux-release.yml:11]: I looks like Reusing workflow can fit https://docs.github.com/en/actions/using-workflows/reusing-workflows?learn=getting_started&learnProduct=actions#calling-a-reusable-workflow

**Review by dshamaev-intermedia — APPROVED** [review]

**Review by inekrasov-intermedia — APPROVED** [review]

**Review by anikitin-intermedia — APPROVED** [review]

**Pull request merged**